File: solutions/go/annalyns-infiltration/1/annalyns_infiltration.go
package annalyn

func CanFastAttack(knightIsAwake bool) bool {
	return !knightIsAwake
}

func CanSpy(knightIsAwake bool, archerIsAwake bool, prisonerIsAwake bool) bool {
	return prisonerIsAwake || knightIsAwake || archerIsAwake
}

func CanSignalPrisoner(archerIsAwake bool, prisonerIsAwake bool) bool {
	return prisonerIsAwake && !archerIsAwake
}

func CanFreePrisoner(knightIsAwake bool, archerIsAwake bool, prisonerIsAwake bool, petDogIsPresent bool) bool {
	if (petDogIsPresent && !archerIsAwake){
		return true
	}
	if(!petDogIsPresent && prisonerIsAwake && !knightIsAwake && !archerIsAwake){
		return true
	}
	return false
}

File: solutions/javascript/annalyns-infiltration/1/annalyns-infiltration.js

export function canExecuteFastAttack(knightIsAwake){
    return !knightIsAwake
}

export function canSpy(knightIsAwake, archerIsAwake, prisonerIsAwake){
    return knightIsAwake || archerIsAwake || prisonerIsAwake
}

export function canSignalPrisoner(archerIsAwake, prisonerIsAwake){
    return prisonerIsAwake && !archerIsAwake
}

export function canFreePrisoner(knightIsAwake, archerIsAwake, prisonerIsAwake, dog){
    if(dog && !archerIsAwake){
        return true
    }
    return prisonerIsAwake && !knightIsAwake && !archerIsAwake
}
